In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier

In [2]:
from sklearn.datasets import load_iris

In [3]:
df2=load_iris()

In [5]:
dir(df2)

['DESCR', 'data', 'feature_names', 'target', 'target_names']

In [6]:
df2.data.shape

(150, 4)

In [9]:
# Split dataset
from sklearn.model_selection import train_test_split
X_train,X_test,Y_train,Y_test= train_test_split(df2.data,df2.target,test_size = 0.3)

In [11]:
print('lenghth of X_train:',len(X_train))
print('lenghth of X_test:',len(X_test))

lenghth of X_train: 105
lenghth of X_test: 45


In [12]:
# define method for test multiple mode
def get_score(model,X_train,X_test,Y_train,Y_test):
    model.fit(X_train,Y_train)
    return(model.score(X_test,Y_test))

In [13]:
# Logistic Regreassion model

get_score(LogisticRegression(),X_train,X_test,Y_train,Y_test)

0.9333333333333333

In [14]:
# SVM model
get_score(SVC(),X_train,X_test,Y_train,Y_test)

0.9777777777777777

In [16]:
# Randome Forest_model

get_score(RandomForestClassifier(n_estimators= 30),X_train,X_test,Y_train,Y_test)

0.9555555555555556

In [23]:
# Apply Kfold Cross_validation tech

from sklearn.model_selection import StratifiedKFold
sk=StratifiedKFold(n_splits=3)


# cross_val_score function

In [34]:

from sklearn.model_selection import cross_val_score
cf1=[]
cf1.append(cross_val_score(LogisticRegression(),df2.data,df2.target))
print(cf1)
print("Average score of LogisticClass for Iris DS with Kfold = 3:",np.average(cf1))

[array([0.96078431, 0.92156863, 0.95833333])]
Average score of LogisticClass for Iris DS with Kfold = 3: 0.9468954248366014


In [52]:
#SVM Model _score
cf2=[]
cf2.append(cross_val_score(SVC(),df2.data,df2.target,cv=10))  #observes that 'CV'  parameter act as tune 
#to improve  accuracy of model, CV : no of fold in dataset
print(cf2)
print("Average score of SVM for Iris DS with Kfold = 3:",np.average(cf2))

[array([1.        , 0.93333333, 1.        , 1.        , 1.        ,
       0.93333333, 0.93333333, 1.        , 1.        , 1.        ])]
Average score of SVM for Iris DS with Kfold = 3: 0.9800000000000001


In [53]:
#Random forest Model _score
cf3=[]
cf3.append(cross_val_score(RandomForestClassifier(n_estimators=70),df2.data,df2.target,cv=20))
print(cf3)
print("Average score of RFS for Iris DS with Kfold = 3:",np.average(cf3))

[array([1.        , 1.        , 0.88888889, 1.        , 1.        ,
       1.        , 0.77777778, 1.        , 1.        , 0.88888889,
       1.        , 0.83333333, 1.        , 1.        , 0.83333333,
       1.        , 1.        , 1.        , 1.        , 1.        ])]
Average score of RFS for Iris DS with Kfold = 3: 0.961111111111111


# Observation : Here SVM model provide accuracy of 98% score in IRIS dataset compare to other models